spark3/src/main/java/org/apache/spark/sql/connector/iceberg/catalog/ExtendedSupportsDelete.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.connector.iceberg.catalog;
+
+import org.apache.spark.sql.connector.catalog.SupportsDelete;
+import org.apache.spark.sql.sources.Filter;
+
+// this should be part of SupportsDelete when merged upstream
+public interface ExtendedSupportsDelete extends SupportsDelete {
```

Comment:
Seems like Spark is about to freeze 3.1 from Monday so it is unfortunate we did not get that into Spark :(

Reply:
I'll submit a PR next week anyway.

Reply:
We can get it in still. This is a small change.

Reply:
And the freeze date under discussion was at least 1 December, so we have some time.